src/resolvers/nodeResolver.ts

```diff
@@ -1,45 +1,73 @@
 import * as Promise from 'bluebird'
 
+import * as _ from 'lodash'
+
 import { Bundle, FileInfo, Resolver } from '../resolver'
 
 export default class NodeResolver implements Resolver {
 	public priority = 0
 	public name = 'NodeJS'
 
-	private hasPackageJson = false
+	private packageJsonContent: Buffer
+	private hasScripts = false
 
 	public entry(file: FileInfo): void {
 		if (file.name == 'package.json') {
-			this.hasPackageJson = true
+			this.packageJsonContent = file.contents
+		} else if (file.name === 'wscript' || _.endsWith(file.name, '.gyp')) {
+			this.hasScripts = true
 		}
 	}
 
 	public isSatisfied(bundle: Bundle): boolean {
-		return this.hasPackageJson
+		return this.packageJsonContent != null
```

Comment:
This suggests `this.packageJsonContent` can be null but it's declared as plain `Buffer`. We either need to add the TS compiler flag that makes strict nullability checks or annotate the member as `Buffer | undefined` (or whatever the syntax is in TS).